### PR TITLE
Draft of error management proposal in route-manager

### PR DESCRIPTION
**Pull request by marcoala**
Readable version of this proposal

https://github.com/festicket/front-end-meta/blob/marcoala/routing-errors/proposals/route-manager-errors.md

### CODE REVIEW

File: proposals/route-manager-errors.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+This will protect us form involuntary changes, since a misconfiguration will trigger error in our testing suite.
+
+But also help us in daily use.
+
+## getPattern
+
+Should check if the pattern has being registered.
+
+```javascript
+router.getUrl('not-a-thing');
```
Comment (mkaszas) [inline proposals/route-manager-errors.md]:
```suggestion
router.getPattern('not-a-thing');
```

File: proposals/route-manager-errors.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+// Error: the patter "not-a-thing" has not being registered
```
Comment (mkaszas) [inline proposals/route-manager-errors.md]:
```suggestion
// Error: the pattern "not-a-thing" has not been registered
```

File: proposals/route-manager-errors.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+//  - the parameter "series" is required in the pattern
+//  - the parameter "edition" is required in the pattern
+```
+
+And also check that the pattern has being registered.
+
+```javascript
+router.getUrl('not-a-thing');
+
+// Error: the patter "not-a-thing" has not being registered
```
Comment (mkaszas) [inline proposals/route-manager-errors.md]:
```suggestion
// Error: the pattern "not-a-thing" has not been registered
```

File: proposals/route-manager-errors.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+Currently our package [route-manager](https://github.com/festicket/route-manager) currently fail silently in a series of scenario.
+I believe that should raise understandable errors if use improperly.
+
+This will protect us form involuntary changes, since a misconfiguration will trigger error in our testing suite.
+
+But also help us in daily use.
+
+## getPattern
+
+Should check if the pattern has being registered.
```
Comment (mkaszas) [inline proposals/route-manager-errors.md]:
```suggestion
Should check if the pattern has been registered.
```

File: proposals/route-manager-errors.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+```javascript
+router.getUrl('festivalGuide', { festivalPk: '123'});
+
+// Error: wrong parameters for the pattern "festivalGuide"
+//  - the parameter "festivalPk" is not present in the pattern
+//  - the parameter "series" is required in the pattern
+//  - the parameter "edition" is required in the pattern
+```
+
+And also check that the pattern has being registered.
```
Comment (mkaszas) [inline proposals/route-manager-errors.md]:
```suggestion
And also check that the pattern has been registered.
```

File: proposals/route-manager-errors.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+```javascript
+router.getPattern('not-a-thing');
+
+// Error: the pattern "not-a-thing" has not been registered
+```
+
+## getUrl
+
+Should checks the parameters, raising errors if the format is not correct.
```
Comment (rlengvenis) [inline proposals/route-manager-errors.md]: check

File: proposals/route-manager-errors.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
+# Error in the route managers
+
+
+Currently our package [route-manager](https://github.com/festicket/route-manager) currently fail silently in a series of scenario.
```
Comment (rlengvenis) [inline proposals/route-manager-errors.md]: Currently two times

Comment (rlengvenis) [reply proposals/route-manager-errors.md]: of scenarious

**Pull request merged**